libmscore/chordrest.h

```diff
@@ -40,13 +40,35 @@ class Spanner;
 enum class SegmentType;
 
 //-------------------------------------------------------------------
-//   @@ ChordRest
+//   @@ ChordRestW
+//   @W ChordRest
 ///    Virtual base class. Chords and rests can be part of a beam
-//
+///   <br>named properties (get/set): <i>small, beam_mode, staff_move, duration_type</i>
 //   @P beamMode      enum (Beam.AUTO, .BEGIN, .MID, .END, .NONE, .BEGIN32, .BEGIN64, .INVALID)
 //   @P durationType  int
 //   @P small         bool           small chord/rest
 //-------------------------------------------------------------------
+class ChordRestW : public DurationElementW {
+      Q_OBJECT
```

Comment:
wouldn't `Q_GADGET` be enough here?

Reply:
Actually, no, not as far as I can tell. Has to be a descendant of QObject to export.
Werner felt so too, if you look at the definitions for what was exposed in the current master branch. they are all QObjects (with a Q_OBJECT annotation)

Reply:
OK, I was just wondering...

Reply:
Just double checked, and ALL the qml interface documentation refers to Q_OBJECT... I'm pretty sure it's why Werner started implementing wrappers in the first place. (I think... I don't pretend to be an expert)

Reply:
Or why he left those at Q_OBJECT and only changed the others to Q_GADGET

Reply:
I did some experiments, and Q_GADGET does NOT play well with Qml. I wonder if it has something to do with 5.8 vs 5.9?